backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/utils/ZonedDateTimeCodec.java

```diff
@@ -0,0 +1,171 @@
+
+package org.apache.james.backends.cassandra.utils;
+
+import static com.datastax.driver.core.ParseUtils.isLongLiteral;
+import static com.datastax.driver.core.ParseUtils.quote;
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.nio.ByteBuffer;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.Date;
+
+import org.apache.james.backends.cassandra.init.CassandraZonedDateTimeModule;
+
+import com.datastax.driver.core.DataType;
+import com.datastax.driver.core.ParseUtils;
+import com.datastax.driver.core.ProtocolVersion;
+import com.datastax.driver.core.TypeCodec;
+import com.datastax.driver.core.UserType;
+import com.datastax.driver.core.exceptions.InvalidTypeException;
+
+public class ZonedDateTimeCodec extends TypeCodec.AbstractUDTCodec<ZonedDateTime> {
+
+    private static final java.time.format.DateTimeFormatter FORMATTER = new java.time.format.DateTimeFormatterBuilder()
+    .parseCaseSensitive()
+    .parseStrict()
+    .append(java.time.format.DateTimeFormatter.ISO_LOCAL_DATE)
+    .optionalStart()
+    .appendLiteral('T')
+    .appendValue(java.time.temporal.ChronoField.HOUR_OF_DAY, 2)
+    .appendLiteral(':')
+    .appendValue(java.time.temporal.ChronoField.MINUTE_OF_HOUR, 2)
+    .optionalEnd()
+    .optionalStart()
+    .appendLiteral(':')
+    .appendValue(java.time.temporal.ChronoField.SECOND_OF_MINUTE, 2)
+    .optionalEnd()
+    .optionalStart()
+    .appendFraction(java.time.temporal.ChronoField.NANO_OF_SECOND, 0, 9, true)
+    .optionalEnd()
+    .optionalStart()
+    .appendZoneOrOffsetId()
+    .optionalEnd()
+    .toFormatter()
+    .withZone(java.time.ZoneOffset.UTC);
+
+    public ZonedDateTimeCodec(UserType definition) {
+        super(definition, java.time.ZonedDateTime.class);
+        checkArgument(definition.getFieldType(CassandraZonedDateTimeModule.DATE).equals(DataType.timestamp()),
+            "Expected UDT with field %s being a timestamp. Got %s",
+            CassandraZonedDateTimeModule.DATE,
+            definition.getFieldType(CassandraZonedDateTimeModule.DATE));
+        checkArgument(definition.getFieldType(CassandraZonedDateTimeModule.TIME_ZONE).equals(DataType.varchar()),
+            "Expected UDT with field %s being a text. Got %s",
+            CassandraZonedDateTimeModule.TIME_ZONE,
+            definition.getFieldType(CassandraZonedDateTimeModule.TIME_ZONE));
+    }
+
+    @Override
+    protected ZonedDateTime newInstance() {
+        return null;
+    }
+
+    @Override
+    protected ByteBuffer serializeField(ZonedDateTime zonedDateTime, String fieldName, ProtocolVersion protocolVersion) {
+        switch (fieldName) {
+            case CassandraZonedDateTimeModule.DATE:
+                return timestamp().serialize(new Date(zonedDateTime.toInstant().toEpochMilli()), protocolVersion);
```

Comment:
Use `bigint().serializeNoBoxing(zonedDateTime.toInstant().toEpochMilli(), protocolVersion)` instead to avoid creating a temporary `Date` object.